### PR TITLE
apply null check

### PR DESCRIPTION
**Pull request by offMomySon**

### CODE REVIEW

File: src/main/java/kr/or/connect/reservation/controller/ReservationApiController.java [hunk 1]
```diff
@@ -51,24 +54,17 @@ public ReservationRequestRs postBook(@RequestBody ReservationRequestRs reservati
 		return rsvMap;
 	}
 	
+	@NonNull
```
Comment (offMomySon) [inline src/main/java/kr/or/connect/reservation/controller/ReservationApiController.java:57]: I don't know if I need to put nonNull on the controller.

File: src/main/java/kr/or/connect/reservation/controller/ReservationApiController.java [hunk 2]
```diff
@@ -51,24 +54,17 @@ public ReservationRequestRs postBook(@RequestBody ReservationRequestRs reservati
...
-				log.debug("Fail to get reservation FROM reservationId = {}", reservationId, e);
-			}
-		}
 
 		return result;
 	}
 	
 	
-	public void storeEmailInfoIfNeeded(List<ReservationResponseRs> responseList, HttpSession session, String reservationEmail) {
+	public void storeEmailInfoIfNeeded(@NonNull List<ReservationResponseRs> responseList,@NonNull HttpSession session,@NonNull String reservationEmail) {
```
Comment (offMomySon) [inline src/main/java/kr/or/connect/reservation/controller/ReservationApiController.java:67]: I don't know if I need to attach it to every param.

File: src/main/java/kr/or/connect/reservation/dto/ReservationResponseRs.java [hunk 3]
```diff
@@ -131,7 +133,7 @@ public DisplayInfoRs getDisplayInfo() {
 		return displayInfoRs;
 	}
 
-	public void setDisplayInfo(DisplayInfoRs displayInfoRs) {
+	public void setDisplayInfo(@NonNull DisplayInfoRs displayInfoRs) {
```
Comment (offMomySon) [inline src/main/java/kr/or/connect/reservation/dto/ReservationResponseRs.java:136]: I am not sure if NonNull is necessary for the set function of the model.
Can you give me some advice?

File: src/main/java/kr/or/connect/reservation/service/impl/ReservationServiceImpl.java [hunk 4]
```diff
@@ -33,12 +36,14 @@
 	@Autowired
 	private ProductPriceRepository pdPriceRep;
 
+	@NonNull
```
Comment (offMomySon) [inline src/main/java/kr/or/connect/reservation/service/impl/ReservationServiceImpl.java:39]: @NonNull 을 method 에 걸었을때,
아래의 request 를 보내서 체크를 해봤지만 예상과는 다르게 에러를 출력하지않습니다
http://localhost:8080/reservation/api/reservations

```	
        @NonNull
	@Override
	@Transactional(readOnly = false)
	public ReservationRequestRs addReservation(@NonNull ReservationRequestRs rsvRequest) {
		setNewDate(rsvRequest);
		ReservationInfo reservationInfo = makeReservationInfo(rsvRequest);

		log.debug("rsv = {}", reservationInfo);

		reservationInfo = rsvRep.save(reservationInfo);

		insertPriceList(reservationInfo.getId(), rsvRequest.getPrices());

		return null;
	}```

File: src/main/java/kr/or/connect/reservation/controller/CategoryApiController.java [hunk 5]
```diff
@@ -9,13 +9,15 @@
 import org.springframework.web.bind.annotation.RestController;
 
 import kr.or.connect.reservation.service.CategoryService;
+import lombok.NonNull;
```
Comment (joey-yoonsung) [inline src/main/java/kr/or/connect/reservation/controller/CategoryApiController.java:12]: Do not use null-related annotation in lombok. 
Please use jsr-305 library in here. https://mvnrepository.com/artifact/com.google.code.findbugs/jsr305

File: src/main/java/kr/or/connect/reservation/dto/ReservationResponseRs.java (hunk 3 above)
Comment (joey-yoonsung) [reply src/main/java/kr/or/connect/reservation/dto/ReservationResponseRs.java:136]: I should be.

File: src/main/java/kr/or/connect/reservation/service/impl/ReservationServiceImpl.java (hunk 4 above)
Comment (joey-yoonsung) [reply src/main/java/kr/or/connect/reservation/service/impl/ReservationServiceImpl.java:39]: Nonnull library 를 바꿔서 해보세요.

File: src/main/java/kr/or/connect/reservation/controller/ReservationApiController.java (hunk 2 above)
Comment (joey-yoonsung) [reply src/main/java/kr/or/connect/reservation/controller/ReservationApiController.java:67]: Use [ParametersAreNonnullByDefault](https://www.javadoc.io/doc/com.google.code.findbugs/jsr305/3.0.1/javax/annotation/ParametersAreNonnullByDefault.html)

**Review by joey-yoonsung — CHANGES_REQUESTED** [review]

File: src/main/java/kr/or/connect/reservation/controller/ReservationApiController.java (hunk 1 above)
Comment (joey-yoonsung) [reply src/main/java/kr/or/connect/reservation/controller/ReservationApiController.java:57]: Do not need.

**Pull request merged**